### ISSUE TITLE
Support for react-native

### ISSUE THREAD

**Issue by guptaash** [PaulAvery/kodi-ws#9]
I was trying to use this library from react-native and notice that websocket dependency is not available. Is it possible to use socket.io rather than ws?

**Comment by PaulAvery** [comment]
Hmm, socket.io would only be able to provide a websocket implementation clientside. Any other transport would not be usable as we do not control the server. I would prefer to avoid the additional overhead. I will update the `ws` dependency soon.

I have never worked with react native, but according to [this](https://facebook.github.io/react-native/docs/network.html#websocket) page there seem to be native websockets available? Not sure how we could make this use those whle still remaining compatability with node, so any input would be great.

**Comment by guptaash** [comment]
Yes, after posting this I also noticed the native ws implementation. Rest of the dependencies should work on react-native as it is. I will also give it a try to see if updating ws to react-native ws would be a simple replacement

**Comment by guptaash** [comment]
I was able to get it working on react-natvie using native implementation -
I had to modify all event handlers and loadSchema function -

Connection.prototype.loadSchema = function() {
	var self = this;
	var fetchSchema = jrpc.run('JSONRPC.Introspect', [], this.socket.send.bind(this.socket));
	this.socket.onmessage = function(message) {
		fetchSchema.handle(message.data);
	}

	return fetchSchema.then(function(schema) {
		//self.socket.removeListener('message', fetchSchema.handle);
		return schema;
	});
};
